tests/pkg_qr-code-generator/Kconfig

```diff
@@ -1,7 +1,6 @@
 config APPLICATION
     bool
     default y
-    imply MODULE_AUTO_INIT_SCREEN if BOARD_HAS_DISPLAY
```

Comment:
Should then this `imply` be moved to `MODULE_TOUCH_DEV` and `MODULE_DISP_DEV`?

Reply:
I looked at this but I don't know how to specify a DEFAULT_MODULE via Kconfig.

Reply:
I think specifying
```
config MODULE_DISP_DEV
    bool "Display device generic API"
    depends on TEST_KCONFIG
    imply MODULE_AUTO_INIT_SCREEN
```
(and something similar for `MODULE_TOUCH_DEV` would work. If the dependencies of `MODULE_AUTO_INIT_SCREEN` are not there (e.g. `MODULE_AUTO_INIT` has been disabled) then it's not enabled. The user can also explicitly say `CONFIG_MODULE_AUTO_INIT_SCREEN=n` if it wants it off.

Another option is to do
```
config MODULE_AUTO_INIT_SCREEN
    bool "Auto-initialize screens"
    default y if MODULE_TOUCH_DEV || MODULE_DISP_DEV
```

In both cases we need to know that [this](https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig) still applies when using menuconfig.

Reply:
Applied your first suggestion (and also rebased to fix a conflict)